system/Database/MySQLi/Connection.php

```diff
@@ -368,11 +368,17 @@ public function escapeLikeStringDirect($str)
     /**
      * Generates the SQL for listing tables in a platform-dependent manner.
      * Uses escapeLikeStringDirect().
+     *
+     * @param string $tableName If $tableName is provided will return only this table if exists.
      */
-    protected function _listTables(bool $prefixLimit = false): string
+    protected function _listTables(bool $prefixLimit = false, string $tableName = ''): string
     {
         $sql = 'SHOW TABLES FROM ' . $this->escapeIdentifiers($this->database);
 
+        if (! empty($tableName)) {
```

Comment:
I still think the "unused" case for `$tableName` parameter should be null, but either way please make this a proper type check. For string-only type this should be:
```suggestion
        if ($tableName !== '') {
```

Reply:
I think we should not use `empty()`.
It is not strict typed.

Reply:
But let your ‘Yes’ be ‘Yes’ and your ‘No’ be ‘No.’
Whatever is more than these is of the evil one

https://en.wikipedia.org/wiki/Matthew_5:37